solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -112,11 +114,42 @@ impl TokenTransferExecutionContext for IbcStorage<'_, '_> {
             amt.denom.trace_path,
             amt.denom.base_denom
         );
-        let amount_in_u64 = check_amount_overflow(amt.amount)?;
+        let store = self.borrow();
+
+        let private_storage = &store.private;
+
+        let asset = private_storage
+            .assets
+            .iter()
+            .find(|asset| {
+                asset.hashed_base_denom.eq(&lib::hash::CryptoHash::digest(
+                    &amt.denom.base_denom.as_str().as_bytes(),
+                ))
+            })
+            .ok_or(TokenTransferError::InvalidToken)?;
+
+        let converted_amount = convert_decimals(
+            &amt.amount,
+            asset.original_decimals,
+            asset.effective_decimals_on_sol,
+        )
```

Comment:
I’m a bit confused.  We have conversion from original decimals to decimals on Solana, but where does conversion in the opposite direction happen? When someone transfers tokens back the conversion must happen in opposite and I don’t see it in the code.

Reply:
well the thing is the amount in the packet always contain with original decimals, it only during mint and burning that we chop them off.